pkg/controller/uniteddeployment/allocator.go

```diff
@@ -0,0 +1,229 @@
+package uniteddeployment
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"k8s.io/klog"
+
+	appsv1alpha1 "github.com/openkruise/kruise/pkg/apis/apps/v1alpha1"
+	"github.com/openkruise/kruise/pkg/controller/uniteddeployment/subset"
+)
+
+type nameToReplicas struct {
+	Name       string
```

Comment:
`Name` here is the workload name which, in this case, is a StatefulSet's name. It is different from `subsetName`. SubsetName is labelled on a workload.

Assuming the subset workload name is in the format '<deployment-name>-<subset-name>' is a good idea, but I realize it is not scalable. For example, it will give up the possibility that `UnitedDeployment` may be able to adopt a subset with an existing workload, because the subset's name maybe not in an expected format. I think it should be an easy thing for ud to adopt a subset. 

As for `subsetName`, we can record it as subset's label, and it will be allowed to be updated if necessary.

If it makes sense, we may also need to update the subsetName [comment](https://github.com/openkruise/kruise/blob/master/pkg/apis/apps/v1alpha1/uniteddeployment_types.go#L85).

Reply:
Adopting workload may hurt usability. It is getting harder to find all the workloads that a ud manages. 
While I see the value of adopting existing sts. But what the rules of adopting existing sts are? Check to see if they have the same StatefulSetTemplateSpec? Are we going to make sure the replica/partition, and node selector settings in ud spec match the existing sts before adopting?

Reply:
ud adopts subset mainly depending on selector and the labels. Once ud adopts a subset, it will sync the replica and partition. And ud will also try to check the spec of new subset up-to-date by comparing the subset label `controller-revision-hash` with ud controller revision. If the subset is out of date, it will be updated. I will commit this part of codes on the following commit.